src/main/java/com/develhope/spring/vehicles/controller/VehicleController.java

```diff
@@ -22,6 +22,16 @@ public ResponseEntity<Vehicle> createVehicle(@RequestBody Vehicle vehicle) {
         return ResponseEntity.ok(vehicleAdded);
     }
 
+    @GetMapping("/user/{userId}")
+    public ResponseEntity<List<Vehicle>> findPurchasedVehiclesByUserId(@PathVariable Long userId) {
```

Comment:
c'è una svista a monte di tutto, cercherò di essere chiaro, sia i veicoli creati che gli ordini creati non hanno uno stato impostato in fase di creazione , quindi di conseguenza tutta la logica a seguire non ha con cosa lavorare, per curiosità hai provato a testare  con Postman ?